PR for Assignment-1

Harshmanitripathi opened the pull request:
PR for Assignment-1

Commit:
Added main class in assignment1

Commit:
Added the all three Packages from my assignment-1 and removed main class as added previously

Harshmanitripathi commented:
PR for assignment-1 raised with neccessary changes.

Commit:
Added the three packages of First Assignment in assignment1 package

PrashantKumarR commented on src/main/java/org/example/assigment1/ExceptionCheck/Validation.java:
```diff
@@ -0,0 +1,64 @@
+
+
+import java.util.*;
+public class Validation extends Exception{
```
Format the code properly
There shouldn't be space at random places
There should be braces around if, else, loop etc even if there is single statement in the. block.
The coding should follow general Java coding guidelines. read the Clean Code book.

PrashantKumarR commented on src/main/java/org/example/assigment1/ExceptionCheck/Validation.java:
```diff
@@ -0,0 +1,64 @@
+        boolean check=true;
+        try {
+
+            for (int i=0;i < valueToValidate.length();i++) {
```
You can simple parse the input into integer, double etc and handle the exception in try catch.

PrashantKumarR commented on src/main/java/org/example/assigment1/ExceptionCheck/Validation.java:
```diff
@@ -0,0 +1,64 @@
+        public static boolean validateForName(String nameOfitem){
+        boolean check = true;
+        try{
+            if (nameOfitem.charAt(0) >= '0' && nameOfitem.charAt(0) <= '9'){
```
Is this a requirement in the Assignment?

Harshmanitripathi replied:
It is checking whether the given name can be considered as name or not.

PrashantKumarR commented on src/main/java/org/example/assigment1/Inventory/CalculateTaxes.java:
```diff
@@ -0,0 +1,33 @@
+
+public class CalculateTaxes {
+
+    public static String calculateTax(String typeOfItem, String priceOfItem, String quantityOfITem) {
```
Why is calculate tax method returning string value? Tax, price or amount is double value.

PrashantKumarR commented on src/main/java/org/example/assigment1/Inventory/CalculateTaxes.java:
```diff
@@ -0,0 +1,33 @@
+
+    public static String calculateTax(String typeOfItem, String priceOfItem, String quantityOfITem) {
+
+        if (typeOfItem.equalsIgnoreCase("Raw")){
```
Write Unit test cases using JUnit to test this method

PrashantKumarR commented on src/main/java/org/example/assigment1/Inventory/UserInputOutput.java:
```diff
@@ -0,0 +1,73 @@
+                } else if (typeOfItem.equalsIgnoreCase("i")) {
+                    typeOfItem = "Imported"; break;
+                }
+                System.out.println("Enter valid input for item type");
```
"Enter valid input for item type" message will be printed even when the input is correct.

Harshmanitripathi replied:
nahi hoga kyoki jab sahi input ayega tho tho uss input ko typeOfItems me assign karne ke baad vo break kar jayega waha se.

PrashantKumarR commented on src/main/java/org/example/assigment1/Inventory/UserInputOutput.java:
```diff
@@ -0,0 +1,73 @@
+
+            }
+            priceOfItem = CalculateTaxes.calculateTax(typeOfItem,priceOfItem,quantityOfItem);
+            ProductDetails productDetails = new ProductDetails(nameOfItem,priceOfItem,quantityOfItem,typeOfItem);
```
Price and tax amount should be stored separately

PrashantKumarR commented on src/main/java/org/example/assigment1/Inventory/UserInputOutput.java:
```diff
@@ -0,0 +1,73 @@
+        showItemsListed();
+    }
+    public static void showItemsListed(){
+        System.out.println("ItemName \t ItemPrice \t Itemquantity \t ItemType");
```
The output should be as per the Assignments
Use String.format() and override toString() methods to print the output

PrashantKumarR commented on src/main/java/org/example/assigment1/Products/ProductDetails.java:
```diff
@@ -0,0 +1,44 @@
+package Products;
+
+public class ProductDetails {
+    String itemName,itemPrice,itemquantity,itemtype;
```
Use proper access modifiers

Commit:
Added the changed asked in almost every file as well as added test cases

Harshmanitripathi committed:
Changes in file:-
1. CalculateTaxes
2. CalculateTaxesTest
3. ProductDetails
4. UserInputOutput
5. Validation

Reason for changes:-
wrong taxed value was coming and input were taken for wrong input.
